ddtrace/opentracer/span.py

```diff
@@ -1,39 +1,157 @@
+import time
 from opentracing import Span as OpenTracingSpan
+from ddtrace.span import Span as DatadogSpan
+from ddtrace.ext import errors
+
+
+class SpanLogRecord(object):
+    """A representation of a log record."""
+
+    slots = ['record', 'timestamp']
+
+    def __init__(self, key_values, timestamp=None):
+        self.timestamp = timestamp or time.time()
+        self.record = key_values
+
+
+class SpanLog(object):
+    """A collection of log records."""
+
+    slots = ['records']
+
+    def __init__(self):
+        self.records = []
+
+    def add_record(self, key_values, timestamp=None):
+        self.records.append(SpanLogRecord(key_values, timestamp))
+
+    def __len__(self):
+        return len(self.records)
+
+    def __getitem__(self, key):
+        if type(key) is int and key < len(self):
+            return self.records[key]
```

Comment:
If you're explicitly expecting an `int` here, it's probably better to raise a [`TypeError`](https://docs.python.org/2/library/exceptions.html#exceptions.TypeError) instead of returning None. Similarly, we should probably raise an [`IndexError`](https://docs.python.org/2/library/exceptions.html#exceptions.IndexError) for out of range values.

Reply:
Alternately, it could also be reasonable to pass key to the `self.records` list unchecked and let that list's exceptions bubble up.